components/director/internal/open_resource_discovery/service.go

```diff
@@ -673,7 +677,16 @@ func (s *Service) processWebhookAndDocuments(ctx context.Context, tx persistence
 	if len(documents) > 0 {
 		log.C(ctx).Info("Processing ORD documents")
 		if err = s.processDocuments(ctx, app.ID, baseURL, documents, globalResourcesOrdIDs); err != nil {
-			log.C(ctx).WithError(err).Errorf("error processing ORD documents: %v", err)
+			ordValidationError, ok := err.(*ORDDocumentValidationError)
+			if ok {
```

Comment:
This could become a one-liner
```suggestion
			if ordValidationError, ok := err.(*ORDDocumentValidationError); ok {
```